crates/holochain_deterministic_integrity/src/lib.rs

```diff
@@ -1,4 +1,43 @@
-//! The Holochain Deterministic Integrity
+//! Holochain Deterministic Integrity (HDI) is Holochain's data model and integrity toolset for
+//! writing zomes.
+//!
+//! The logic of a Holochain DNA can be divided into two parts: integrity and coordination.
+//! Integrity is the part of the hApp that defines the data structure and validates data
+//! types and values. Coordination encompasses the domain logic and implements the functions
+//! to manipulate data.
+//!
+//! # Examples
+//!
+//! An example of an integrity zome with data definition and data validation can be found in the
+//! wasm workspace of the Holochain repository:
+//! <https://github.com/holochain/holochain/blob/develop/crates/test_utils/wasm/wasm_workspace/integrity_zome/src/lib.rs>.
+//!
+//! # Data definition
+//!
+//! The DNA's data model is defined in integrity zomes. They comprise all data type definitions
+//! as well as relationships between those types. Integrity zomes are purely definitions and do
+//! not contain functions to manipulate the data. Therefore a hApp's data model is encapsulated
```

Comment:
Integrity zomes can't contain other zome functions?

Reply:
@guillemcordoba They can contain other functions, but if you change those functions the DNA hash changes and it defeats one of the purposes to separate integrity from coordinator zomes. That's how I understood it. I may be wrong.

Reply:
I think the difference is that they can contain functions but they can't be exposed as zome functions. Indeed, it might be an interesting design pattern to expose boilerplate functions that coordinators could use for CRUD, though that does raise the risk that Jost talks about.